pkg/install/install.go

```diff
@@ -221,10 +223,34 @@ func (i *Installer) runSteps(ctx context.Context, steps []interface{}) error {
 		switch step := step.(type) {
 		case action:
 			i.log.Printf("running step %s", runtime.FuncForPC(reflect.ValueOf(step).Pointer()).Name())
-			err = step(ctx)
+
+			func() {
+				timeoutCtx, cancel := context.WithTimeout(ctx, 10*time.Minute)
+				defer cancel()
+
+				wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
+					err = step(ctx)
+					if azureerrors.HasAuthorizationFailedError(err) || azureerrors.HasLinkedAuthorizationFailedError(err) {
+						i.log.Print(err)
+						// https://github.com/Azure/ARO-RP/issues/541: it is unclear if this refresh helps or not
+						if development, ok := i.env.(env.Dev); ok {
+							err = development.RefreshFPAuthorizer(ctx, i.fpAuthorizer)
+							if err != nil {
+								return false, err
+							}
+						}
+						return false, nil
+					}
+					return err == nil, err
+				}, timeoutCtx.Done())
+			}()
```

Comment:
it's better in one place, i'd just prefer it gone and re-run create

Reply:
I agree on the goal, but not the means.  I'd like to bring the create steps that are outside the installer steps framework into that framework.

We get so many authorization errors I'm loathe to restart from the start of the phase each time.

Reply:
Maybe restart with backoff timeout?